app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -677,10 +677,15 @@ public function getPublishableKeyForRoute()
         $checkoutType = static::CHECKOUT_TYPE_MULTI_PAGE;
         if ($routeName === 'adminhtml') {
             $checkoutType = static::CHECKOUT_TYPE_ADMIN;
-        } else if ( ($routeName === 'firecheckout') || ($routeName === 'checkout' && $controllerName === 'onepage') ) {
+        } else if ($routeName === 'checkout' && $controllerName === 'onepage') {
             $checkoutType = static::CHECKOUT_TYPE_ONE_PAGE;
         }
 
+        $checkoutType =   $this->boltHelper()->doFilterEvent(
+            'bolt_boltpay_filter_checkout_type',
+            $checkoutType
+        );
```

Comment:
@YevhenShBolt You should give the filter some contextual additional parameters.  In the past code, you constructed a request path, and that seems appropriate.  However, to illustrate the concept using the now existing code, since you used `$routeName` and `$controllerName` as the basis of our standard logic, reason dictates that these same values will be extremely important for someone who needs to use this filter to override the behavior.  Therefore, you should include them in an associative array as the third parameter.